packages/profiles/src/abstract/AbstractProfileManager.ts

```diff
@@ -470,11 +471,20 @@ export abstract class AbstractProfileManager<T extends IProfileTypeConfiguration
       this.log.debug(`The default profile for type "${this.profileType}" is "${parms.name}".`);
 
       // If we don't find the default name and we know fail not found is false, then return here
-      if (isNullOrUndefined(parms.name) && !parms.failNotFound) {
-        return this.failNotFoundDefaultResponse("default was requested");
-      } else if (isNullOrUndefined(parms.name)) {
-        this.log.error(`No default profile exists for type "${this.profileType}".`);
-        throw new ImperativeError({msg: `No default profile set for type "${this.profileType}".`});
+      if (isNullOrUndefined(parms.name)) {
```

Comment:
`isNullOrUndefined` is a deprecated function. The preferred way of checking is `parms.name == null`. Note the double equals `==`.